Add target SPRACINGF7DUAL

sdirkwinkel opened the pull request:
This pull request imports the SPRACINGF7DUAL from Betaflight and gets it building. I've separated importing and fixing into different commits for better accountability.

This currently does not work for me: After flashing my board doesn't get a serial connection in the configurator. I'll propably get a debugger at some point to see what is happening. The board works fine with Cleanflight.

I'd appreciate any pointers as to where to start looking for mistakes as I'm new to inav(and cf/bf).

Ping @hydra: consider removing inav from your description until this works

sdirkwinkel committed:
SPRACINGF7DUAL: import from bf

sdirkwinkel committed:
SPRACINGf7DUAL: port makefile and timers

sdirkwinkel committed:
SPRACINGF7DUAL: add changes for inav

sdirkwinkel committed:
SPRACINF7DUAL: initialize gyro

The SPRACINGF7DUAL has to MPU6500 gyros on one spi bus.
To use one of them we have to initialize both cs pins.

sdirkwinkel committed:
SPRACINGF7DUAL: use TIM_USE_MC_MOTOR

sdirkwinkel committed:
SPRACINGF7DUAL: set gyro alignment

sdirkwinkel committed:
SPRACINGF7DUAL: fix barometer

sdirkwinkel committed:
SPRACINGF7DUAL: fix adc

sdirkwinkel committed:
SPRACINGF7DUAL: remove unused code/cleanup

digitalentity commented on src/main/target/SPRACINGF7DUAL/target.h:
```diff
@@ -0,0 +1,232 @@
+#define TARGET_IO_PORTD         (BIT(2))
+
+//#define USABLE_TIMER_CHANNEL_COUNT 16 // 4xPWM, 8xESC, 2xESC via UART3 RX/TX, 1xLED Strip, 1xIR.
+#define USABLE_TIMER_CHANNEL_COUNT 16
```
This is no longer needed

digitalentity commented on src/main/target/SPRACINGF7DUAL/target.h:
```diff
@@ -0,0 +1,232 @@
+// #define SDCARD_SPI_FULL_SPEED_CLOCK_DIVIDER     4 // 54Mhz/3 = 18Mhz // FIXME currently running slower than 18mhz for testing.
+
+// #define SDCARD_DMA_STREAM_TX_FULL           DMA1_Stream7
+#define SDCARD_DMA_TX                       DMA1
```
SDCARD definitions need refactoring. A lof of defines is not needed:

```
#define SPI3_CLOCK_LEADING_EDGE

#define USE_SDCARD
#define USE_SDCARD_SPI
#define SDCARD_SPI_BUS          BUS_SPI3
#define SDCARD_CS_PIN           PC3
```

digitalentity commented on src/main/target/SPRACINGF7DUAL/target.h:
```diff
@@ -0,0 +1,232 @@
+
+#define MAG_I2C_BUS             BUS_I2C1
+#define USE_MAG
+#define USE_MAG_HMC5883
```
Please add mode supported compasses, at least QMC5883

digitalentity commented on src/main/target/SPRACINGF7DUAL/target.c:
```diff
@@ -0,0 +1,84 @@
+#include "drivers/bus.h"
+
+// Register second MPU6500 so cs pin is initialized
+BUSDEV_REGISTER_SPI(busdev_mpu6500_2,     DEVHW_MPU6500,      MPU6500_2_SPI_BUS,    MPU6500_2_CS_PIN,     GYRO_2_EXTI_PIN,  DEVFLAGS_NONE);
```
It's better to decrare both gyros here and add `USE_TARGET_IMU_HARDWARE_DESCRIPTORS` definition to `target.h`

digitalentity reviewed (changes requested):
Please address comments and rebase as this branch is no longer up to date with `development`. After that this could be merged.

sdirkwinkel committed:
SPRACINGF7DUAL: refactoring

- Update SDCARD definitions
- remove unused defines
- Add more supported compasses
- declare both gyros manually
